.github/workflows/detect-breaking-change.yml

```diff
@@ -0,0 +1,24 @@
+name: "Detect Breaking Changes"
+on: [push, pull_request]
```

Comment:
Wondering how useful is `push` here. 
Also should this be `pull_request_target`? Similar to https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/check-compatibility.yml#L5

Reply:
Good question - pull request checks can be bypassed so the results **might** not be used [1].  I'd like like a record of every breaking change on a commit by commit bias.

`pull_request_target` is required for workflows that need elevated privileges' such write operations on GitHub repository, whereas this workflow does need any additional extra privileges.

- [1] They can be required, but it removes agency from maintainers.  I would rather maintainers have the flexibility and only if the maintainers agree should we make changes https://github.com/opensearch-project/OpenSearch/issues/12735